// educaid-react/src/services/reportApi.js
export async function previewReport(formData) {
  const form = new URLSearchParams();
  Object.entries(formData || {}).forEach(([k, v]) => {
    if (v !== undefined && v !== null) form.append(k, String(v));
  });
  form.set('action', 'preview');

  const response = await fetch('/api/reports/generate_report.php', {
    method: 'POST',
    credentials: 'include',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
      'X-Requested-With': 'XMLHttpRequest',
    },
    body: form.toString(),
  });

  const data = await response.json().catch(() => ({}));
  return { ok: response.ok, status: response.status, data };
}

export async function getReportStatistics(query = {}) {
  const params = new URLSearchParams({ action: 'get_statistics' });
  Object.entries(query || {}).forEach(([k, v]) => {
    if (v !== undefined && v !== null) params.append(k, String(v));
  });

  const response = await fetch(`/api/reports/generate_report.php?${params.toString()}`, {
    method: 'GET',
    credentials: 'include',
    headers: {
      'X-Requested-With': 'XMLHttpRequest',
    },
  });

  const data = await response.json().catch(() => ({}));
  return { ok: response.ok, status: response.status, data };
}
